radiospectra/spectrogram2/spectrogram.py

```diff
@@ -671,6 +674,79 @@ def _read_cdf(file):
                 "freqs": freqs,
             }
             return data, meta
+        elif "SOLO" in cdf_globals.get("Project", ""):
+            if "RPW-HFR-SURV" not in cdf_globals.get("Descriptor", ""):
+                raise ValueError(
+                    f"Currently radiospectra supports Level 2 HFR survey data the file"
+                    f'{file.name} is {cdf_globals.get("Descriptor", "")}'
+                )
+
+            # Extract variables
+            all_times = cdf.varget("EPOCH")
+            all_freqs = cdf.varget("FREQUENCY")
+            sensor = cdf.varget("SENSOR_CONFIG")
+            freq_uniq = np.unique(cdf.varget("FREQUENCY"))
+
+            acg1 = cdf.varget("AGC1")
+            acg2 = cdf.varget("AGC2")
+
+            def extract_data(acg, freq_uniq, sensors, sensor_num, sensor_value):
+                slices = []
+                times = []
+                freqs = []
+                for cfreq in freq_uniq:
+                    search = np.argwhere((all_freqs == cfreq) & (sensors[:, sensor_num] == sensor_value) & (acg != 0))
+                    if search.size > 0:
+                        slices.append(acg[search])
+                        times.append(all_times[search])
+                        freqs.append(cfreq)
+                return slices, times, freqs
+
+            sensor_unique = np.unique(sensor)
```

Comment:
is this the correct thing to be indexing? 

Is the idea to find CH1 and CH2 here?